test/new-e2e/tests/otel/otel-agent/infraattributes_eks_test.go

```diff
@@ -40,17 +40,12 @@ agents:
           value: 'disable_operation_and_resource_name_logic_v2'
 `
 	t.Parallel()
-	e2e.Run(t, &iaEKSTestSuite{}, e2e.WithProvisioner(awskubernetes.EKSProvisioner(
-		awskubernetes.WithEKSOptions(
-			eks.WithLinuxNodeGroup(),
-			eks.WithUseAL2023Nodes(),
-		),
-		awskubernetes.WithAgentOptions(
-			kubernetesagentparams.WithHelmValues(values),
-			kubernetesagentparams.WithOTelAgent(),
-			kubernetesagentparams.WithOTelConfig(iaConfig),
-		),
-	)))
+	e2e.Run(t, &iaEKSTestSuite{}, e2e.WithProvisioner(
+		proveks.Provisioner(proveks.WithRunOptions(
+			eks.WithEKSOptions(eks.WithLinuxNodeGroup()),
+			eks.WithAgentOptions(kubernetesagentparams.WithHelmValues(values), kubernetesagentparams.WithOTelAgent(), kubernetesagentparams.WithOTelConfig(iaConfig)),
```

Comment:
nit: keep previous style
```suggestion
			eks.WithAgentOptions(
				kubernetesagentparams.WithHelmValues(values),
				kubernetesagentparams.WithOTelAgent(),
				kubernetesagentparams.WithOTelConfig(iaConfig)
			),
```